d2core/d2render/ebiten/ebiten_surface.go

```diff
@@ -222,6 +239,15 @@ func (s *ebitenSurface) DrawLine(x, y int, fillColor color.Color) {
 	)
 }
 
+// DrawSurface draws a surface within the surface
+func (s *ebitenSurface) DrawSurface(sfc d2interface.Surface) {
+	if _, ok := sfc.(*ebitenSurface); !ok {
+		return
+	}
+
+	sfc.Render(s)
+}
+
 // DrawRect draws a rectangle
```

Comment:
We want to add a `DrawSubSurface` method to `d2interface.Surface` and `ebitenSurface`. The `ebitenSurface` should be utilizing the `SubImage` method that was added recently.

Reply:
same here, it's some rest of junk